python/graphstorm/gsf.py

```diff
@@ -617,8 +623,21 @@ def create_builtin_edge_decoder(g, decoder_input_dim, config, train_task):
     loss_func: The loss function(s)
     """
     dropout = config.dropout if train_task else 0
+
+
     if config.task_type == BUILTIN_TASK_EDGE_CLASSIFICATION:
         num_classes = config.num_classes
+
+        # Focal loss expects 1-dimensional output
+        if config.class_loss_func == BUILTIN_CLASS_LOSS_FOCAL:
+            assert num_classes in [1, 2], (
```

Comment:
Add a comment here:
`For backward compatibility, we allow the num_classes to be 1. We suggest users to set it to 2. `